README.md

```diff
@@ -21,44 +20,133 @@ This Connector implements the [Data Connector Spec](https://github.com/hasura/nd
+
+3. Modify the Docker container port in `my_subgraph/connector/my_openapi/docker-compose.my_openapi.yaml`. Typically, connectors default to port 8080. Each time you add a connector, please increment the published port by one to avoid port collisions. For example:
+```
+ports:
+  - mode: ingress
+    target: 8080
+    published: '8082'
+    protocol: tcp
+```
+
+4. Introspect the OpenAPI document using the connector
+
+```
+ddn connector introspect --connector my_subgraph/connector/my_openapi/connector.yaml
+```
+
+This will introspect your OpenAPI document and create an `api.ts` file, a `functions.ts` file and other supporting files required to run the Typescript project.
+
+- The `api.ts` file contains the Data Types and API calls from the OpenAPI document
+- The `functions.ts` file contains functions that wrap API calls. You can modify this `functions.ts` file to introduce business logic if you want to. See [Saving User Changes](#saving-user-changes) if you want to preserve your changes in this file when you introspect the OpenAPI document again.
+
+5. Add a [Data Connector Link](https://hasura.io/docs/3.0/supergraph-modeling/data-connector-links)
+
+```
+ddn connector-link add my_openapi --subgraph my_subgraph
+```
+
+This will create a file `my_subgraph/metadata/my_openapi.hml` that links your OpenAPI Connector to your Hasura Supergraph.
+
+6. Update the evironment variables listed in `my_subgraph/metadata/my_openapi.hml` (here, `MY_SUBGRAPH_MY_OPENAPI_READ_URL` and `MY_SUBGRAPH_MY_OPENAPI_WRITE_URL`) in `my_subgraph/.env.my_subgraph`.
+
+7. Start the connector in a Docker container
+
+```
+docker compose -f my_subpgraph/connector/my_openapi/docker-compose.my_openapi.yaml up
+```
+
+The `http://localhost:${your-docker-container-port}/schema/` should return the schema of your OpenAPI Connector. All functions that wrap `GET` requests will be listed in the `functions` array, while all other functions will be listed in the `procedures` array.
+
+8. Add the connector schema to [Data Connector Link](<(https://hasura.io/docs/3.0/supergraph-modeling/data-connector-links)>)
+
+```
+ddn connector-link update my_openapi --subgraph my_subgraph
+```
+
+This command will modify `my_subgraph/metadata/my_openapi.hml`, and the schema of the connector will be added to the `definition.schema` key.
+
+9. Add all resources in your connector schema to your GraphQL API
+
+```
+ddn connector-link update my_openapi --subgraph my_subgraph --add-all-resources
+```
+
+This will create HML files that represent resources in your connector schema at `my_subgraph/metadata`. These HML files specify your GraphQL API.
+
+You have now added the OpenAPI Connector and imported all of your APIs in your supergraph. You can now:
+
+- Deploy the supergraph to Hasura DDN (Please follow the steps [here](https://hasura.io/docs/3.0/getting-started/deployment/))
```

Comment:
Following the above doc leads to https://hasura.io/docs/3.0/getting-started/deployment/deploy-a-connector which if I click on "Other connectors" leads back to this doc. I suggest we clearly call out what we need to deploy this connector on DDN right here itself till we figure this out.

Reply:
@rsd1122  I'm sorry I don't quite understand what's happening for you here. Is that https://hasura.io/docs/3.0/getting-started/deployment/ always gets redirected to https://hasura.io/docs/3.0/getting-started/deployment/deploy-a-connector?

Reply:
@m-Bilal yes - its a circular dependency.

Reply:
Currently, it looks like this goes to the correct documentation for deploying a connector.